### PR TITLE
Restore snippets

### LINKED ISSUE #4: Changes in Gmail cause firefox to be crashed by Gmail Manager
Started from yesterday, again, firefox froze when gmail manager was checking the email account. This kind of situation happened before. As I remember, it was the reason people started making a fork from Gmail Manager by Todd Long. It seems the old problem was triggered by some recent changes in gmail. It happened with both Gmail Manager NG and Gmail Manager-community. From the reports in the gmail manager thread in mozilla forum, people using different versions of firefox observed same issue.

### PR DESCRIPTION
**Pull request by ath0mas**
Cleanup commented out failing regex for snippets.
Review similar data matching for quota and counters/labels

To close #4 :sparkles:

### CODE REVIEW

**Comment by ath0mas** [comment]
Please test pre-release [0.6.5.3-3.beta.1](https://github.com/gmail-manager-community/gmail-manager-community/releases/tag/0.6.5.3-3.beta.1)

File: src/components/gmServiceGmail.js [hunk 1]
```diff
@@ -557,46 +557,49 @@ gmServiceGmail.prototype = {
           }
           
           try {
-            //var tbMatches = data.match(/\["tb",(?:.|\s)+?](?:\s]){2,}(?!\s,,)/g);   see: https://github.com/gmail-manager-community/gmail-manager-community/issues/4
-            var tbMatches = [];
+            var loc1 = data.indexOf("var VIEW_DATA=[[");
+            var loc2 = data.lastIndexOf("var GM_TIMING_END_CHUNK2");
+            var viewData = data.substring(loc1+14, loc2-2);
```
Comment (ath0mas) [inline src/components/gmServiceGmail.js:563]: @arix98 can you explain why +14 and -2?

and is there a way to get full data like fetched by addon? a URL or so?
I thought it was https://mail.google.com/mail/u/0/feed/atom, but seems not to be...

Comment (grewt) [inline src/components/gmServiceGmail.js:563]: @ath0mas, data.indexOf(string) returns the start position of string in data, so:
- after finding the start of `"var VIEW_DATA=[["`, move forward 14 characters to the start of the JSON (the first of the 2 `[` in that string)
- after finding `"var GM_TIMING_END_CHUNK2"`, go back 2 characters to where the JSON ends

These offsets could have been in the `var locx =` lines, but it's the same end result.

Comment (ath0mas) [inline src/components/gmServiceGmail.js:563]: Ok fine, it's what I thought... but I wasn't sure how the JSON string ended here

**Comment by ath0mas** [comment]
Now with pre-release [0.6.5.3-3.beta.2](https://github.com/gmail-manager-community/gmail-manager-community/releases/tag/0.6.5.3-3.beta.2)

**Comment by ath0mas** [comment]
Can you @damienfournier and/or @arix98 test if the '&' in labels problem is fixed, or not?

**Comment by M3Rocket** [comment]
@ath0mas The Amperstand "&" seems fixed to me. At least as far as I can tell. All my snippets display properly regardless of whether there's a "&" in the subject or content. 

Edit: Ooops I think after re-reading https://github.com/gmail-manager-community/gmail-manager-community/issues/4 the "&" affects the "New email" count? If so, on my setup at least, the count is correct even with "&" in the label/subject/snippet content.

**Comment by ath0mas** [comment]
@M3Rocket no link with "New email" count when I read #4

**Comment by damienfournier** [comment]
@M3Rocket the "&" in a label was affecting the number of unread mail on the icon of gmail manager (set to 0)

@ath0mas I've tried the fix on my modified version of gmail manager ng, it fixed it.
I've not tried your version, but it should work the same.

If you revert the line about the "&" and you had one in one of your label, do you see it fails when gmail manager login to the account ?

**Comment by arix00** [comment]
See my new comment in #4 

`&` does not break anything in email title since they're encoded in unicode sequences there(\u####), however its in hex sequence(\x##) when used in labels and breaks JSON, I posted a new regex to change them(`& < >` and maybe others) to unicode sequences.

**Comment by arix00** [comment]
@ath0mas that's some quick update! with the new regex, `replace(/\\x26/g, "\u0026")` can be removed now.

File: src/components/gmServiceGmail.js (hunk 1 above)
Comment (arix00) [inline src/components/gmServiceGmail.js:563]: @ath0mas: 
for debug, the data is basically gmail webpage's "view source" -- there may be some slight difference, its a very long string(about 260K here) and got truncated in firefox's output console.

**Comment by ath0mas** [comment]
Test pre-release [0.6.5.3-3.beta.3](https://github.com/gmail-manager-community/gmail-manager-community/releases/tag/0.6.5.3-3.beta.3) please

**Pull request merged**